### ISSUE TITLE
Compiled-introspection checks silently no-op (pipeline drops returned issues)

### ISSUE THREAD

**Issue by zimt28** [leonqadirie/ash_credo#187]
I could not get the checks running that require a compiled project – even obvious violations are not reported while regular AST-based checks work fine. I'm not sure if this is an actual bug or a setup issue on my side but I've followed the instructions in the Readme exactly.
<details>
<summary>As I was not able to solve this issue I asked Claude. The following is its take on it, maybe that helps.</summary>
Environment: ash_credo 0.15.0 · credo 1.7.19 · ash 3.29.3 · Elixir 1.20.2 / OTP 29

Summary
All compiled-introspection checks (Design.MissingTimestamps, Design.MissingPrimaryAction, Design.MissingCodeInterface, Design.MissingIdentity, Warning.AuthorizerWithoutPolicies, Refactor.UseCodeInterface, …) produce no issues when run via mix credo, even when the resource clearly violates them. No config diagnostic, no "could not load" issue, no "Ash unavailable" diagnostic — just silence. The project is compiled beforehand and the checks are enabled in .credo.exs (plugin registered via plugins: [{AshCredo, []}]).

The check logic is correct — credo's pipeline drops the result
From inside a custom check running in the same credo VM, I verified step by step against a postgres-backed resource with no timestamps():
- Code.ensure_loaded?(Ash.Resource.Info) → true
- AshCredo.Introspection.Compiled.ash_available?() → true
- AshCredo.Introspection.Compiled.attributes(MyResource) → {:ok, [...]} (fresh, no timestamp attrs)
- AshCredo.Introspection.resource_contexts(source_file) → 1 context, has_data_layer? true
- AshCredo.Check.Design.MissingTimestamps.run(source_file, params) called directly → returns [%Credo.Issue{message: "Resource is missing timestamps."}] ✅

Yet the same check, invoked through credo's normal run, yields nothing. Same pattern confirmed for MissingPrimaryAction (direct: 1 issue) and MissingCodeInterface (direct: 12 issues) — all invisible via mix credo.

Repro
1. Ash project, postgres-backed resource without timestamps().
2. Enable AshCredo.Check.Design.MissingTimestamps, plugins: [{AshCredo, []}].
3. mix compile && mix credo --strict (also tried single invocation mix do compile + credo).
4. Expected: "Resource is missing timestamps." → Actual: found no issues.

Notes
- Not a compile-ordering problem: the documented "compile before credo" / lint alias remedy does not help; a single mix do compile + credo invocation behaves identically.
- AST-based AshCredo checks (e.g. EmptyDomain) work fine in the same run.
- Smells like issues returned by these checks are lost in result collection/filtering (parallel workers / cache) rather than a check-logic or introspection bug.

</details>

**Comment by leonqadirie** [comment]
Oh, I’ll investigate - please bear some days with me due to current capacity.

**Comment by leonqadirie** [comment]
@zimt28 
I can't reproduce this. Can you provide more information or a MRE?
Please do the following and report back:
```elixir
# A) Point Credo directly at one relevant file.
mix credo lib/my_app/.../that_resource.ex --strict
# if it reports, the the check + file are fine; the normal run is excluding the file
# if it does not report, the check is likely not being dispatched, maybe a configuration issue

# B) Is the check actually active in the resolved config?
mix credo info --verbose | grep MissingTimestamps
# if it is absent, it is not enabled in the config that run resolves (wrong config name / plugin registration)
```
